experanto/datasets.py

```diff
@@ -438,11 +447,219 @@ def __getitem__(self, idx) -> dict:
+            )
+        self._valid_indices = valid_indices
+        # Return the swap times that satisfy all conditions
+        return self._experiment.devices['screen'].timestamps[valid_indices]
+
+
+    def shuffle_valid_screen_times(self) -> None:
+        times = self._full_valid_sample_times
+        self._valid_screen_times = np.sort(np.random.choice(times, size=len(times) // self.sample_stride, replace=False))
+    
+
+    @staticmethod
+    def get_batch_aligned_split(n_images, train_frac, batch_size, seed):
+        if seed is not None:
+            np.random.seed(seed)
+
+        indices = np.arange(n_images)
+        np.random.shuffle(indices)
+
+        # Compute training size and align up to the next batch
+        train_size = int(n_images * train_frac)
+        train_size = ((train_size + batch_size - 1) // batch_size) * batch_size  # Align up
+        train_size = min(train_size, n_images)  # Ensure it doesn't exceed total images
+
+        train_indices = indices[:train_size]
+        val_indices = indices[train_size:]
+        return train_indices, val_indices
+
+    def split(self, train_frac=0.8, seed=None):
+        """
+        Split the dataset into train and validation subsets using the same logic as the static loader.
+        """
+        # Get train and validation indices using the static loader's logic
+        train_indices, val_indices = get_validation_split(len(self), train_frac, seed)
+
+        # Create train and validation subsets
+        train_dataset = Subset(self, train_indices)
+        val_dataset = Subset(self, val_indices)
+        return train_dataset, val_dataset
+
+    def __len__(self):
+        return len(self._swap_times_valid_trial)
+
+    def __getitem__(self, idx) -> dict:
+        out = {}
+        s = self._swap_times_valid_trial[idx]  # Select the swap time for this index
+        for device_name in self.device_names:
+            sampling_rate = self.sampling_rates[device_name]
+            chunk_size = self.chunk_sizes[device_name]
+            chunk_s = chunk_size / sampling_rate
+            times = np.linspace(s, s + chunk_s, chunk_size, endpoint=False)
+            times = times + self.modality_config[device_name].offset
+            if device_name == "screen":
+                (data,meta), _ = self._experiment.interpolate(times, device=device_name)
+                out['image_id'] = meta
+            else:
+                data, _ = self._experiment.interpolate(times, device=device_name)
+            out[device_name] = self.transforms[device_name](data).squeeze(0)
+            
+        out["timestamps"] = torch.from_numpy(times)
+        datapoint = {'screen':out['screen'], 'responses':out['responses']}
+        return out
+        #return self.DataPoint(**datapoint)
+
+
+class MonkeyFixationGazeCrop(MonkeyFixation):
+    def __init__(
+        self,
+        root_folder: str,
+        global_sampling_rate: None,
+        global_chunk_size: None,
```

Comment:
Same here, you probably want sth like [this](https://github.com/sensorium-competition/experanto/blob/700234ca51d82223484f259ba433094e1d255100/experanto/interpolators.py#L79), no?

```suggestion
        global_sampling_rate: typing.Optional[float] = None,
        global_chunk_size: typing.Optional[float] = None,
```

(would require `import typing` before)